metadata-ingestion/tests/unit/test_redshift_lineage.py

```diff
@@ -149,3 +191,587 @@ def test_cll():
+
+
+def test_collapse_temp_recursive_cll_lineage():
+    lineage_extractor = get_lineage_extractor()
+
+    temp_table: TempTableRow = TempTableRow(
+        transaction_id=126,
+        query_text="CREATE TABLE #player_price distkey(player_id) AS SELECT player_id, SUM(price_usd) AS price_usd "
+        "from #player_activity_temp group by player_id",
+        start_time=datetime.now(),
+        session_id="abc",
+        create_command="CREATE TABLE #player_price",
+        parsed_result=SqlParsingResult(
+            query_type=QueryType.CREATE,
+            in_tables=[
+                "urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_activity_temp,PROD)"
+            ],
+            out_tables=[
+                "urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_price,PROD)"
+            ],
+            debug_info=SqlParsingDebugInfo(),
+            column_lineage=[
+                ColumnLineageInfo(
+                    downstream=DownstreamColumnRef(
+                        table="urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_price,PROD)",
+                        column="player_id",
+                        column_type=SchemaFieldDataTypeClass(NumberTypeClass()),
+                        native_column_type="INTEGER",
+                    ),
+                    upstreams=[
+                        sqlglot_l.ColumnRef(
+                            table="urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_activity_temp,PROD)",
+                            column="player_id",
+                        )
+                    ],
+                    logic=None,
+                ),
+                ColumnLineageInfo(
+                    downstream=DownstreamColumnRef(
+                        table="urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_price,PROD)",
+                        column="price_usd",
+                        column_type=SchemaFieldDataTypeClass(NumberTypeClass()),
+                        native_column_type="BIGINT",
+                    ),
+                    upstreams=[
+                        sqlglot_l.ColumnRef(
+                            table="urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_activity_temp,PROD)",
+                            column="price_usd",
+                        )
+                    ],
+                    logic=None,
+                ),
+            ],
+        ),
+        urn="urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_price,PROD)",
+    )
+
+    temp_table_activity: TempTableRow = TempTableRow(
+        transaction_id=127,
+        query_text="CREATE TABLE #player_activity_temp SELECT player_id, SUM(price) AS price_usd "
+        "from player_activity",
+        start_time=datetime.now(),
+        session_id="abc",
+        create_command="CREATE TABLE #player_activity_temp",
+        parsed_result=SqlParsingResult(
```

Comment:
I would've liked this test to be more broad. For example, we should be mocking the query results from the query that fetches temp tables and relying on the redshift lineage implementation to parse build `TempTableRow` instances. That way, we would not have any hardcoded `SqlParsingResult` objects in our tests and would ensure that we can actually fetch temp table definitions correctly

Reply:
I mock the query result for `test_collapse_temp_lineage`.

Reply:
Is there a reason you didn't make similar changes to the rest of the tests?